### PR TITLE
Replace sprintf, %, and .format with f-strings 

### PR DESCRIPTION
**Pull request by eumiro**
Now this is a draft of continuation of #68, so it includes all its commits and I will rebase it when the other one gets merged.

*It works on my computer™*, but I want to check it with GHA CI before I leave it for a review.

### CODE REVIEW

**Comment by eumiro** [comment]
Okay, now it works and I made it independent from the previous PR. When @Lucas-C you merge #68, I'll rebase this again. Please feel free to review.

**Comment by eumiro** [comment]
@Lucas-C rebased, repushed. :rocket:

**Comment by Lucas-C** [comment]
@eumiro I'd be happy to merge your code once you have addressed the only issue I found in your PR, mentioned above 😉

File: fpdf/fpdf.py [hunk 1]
```diff
@@ -2161,10 +2087,10 @@ def _enddoc(self):
         with self._trace_size("xref"):
             o = len(self.buffer)
             self._out("xref")
-            self._out("0 " + (str(self.n + 1)))
+            self._out(f"0 {self.n + 1}")
             self._out("0000000000 65535 f ")
             for i in range(1, self.n + 1):
-                self._out(sprintf("%010d 00000 n ", self.offsets[i]))
+                self._out(f"{self.offsets[i]:010d} 00000 n ")
```
[the remark this answers is not in the document]
Comment (eumiro) [reply fpdf/fpdf.py]: Actually no, as well as `:s}` is not needed. So I added another commit removing these formatters. If it's green and you like it, you can squash both commits and leave only the message from the first one.

**Review by Lucas-C — APPROVED** [review]

**Pull request merged**